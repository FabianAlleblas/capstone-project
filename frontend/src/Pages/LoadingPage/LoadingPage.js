import LoadingSpinner from '../../Components/LoadingSpinner/LoadingSpinner'
import { ReactComponent as AppLogo } from '../../assets/logos/urbanplantslogo.svg'
import styled from 'styled-components/macro'

export default function LoginPage() {
  return (
    <PageWrapper>
      <AppLogo />
      <LoadingSpinner />
    </PageWrapper>
  )
}

const PageWrapper = styled.section`
  background-color: var(--primary-dark);
  display: grid;
  height: 100vh;
  place-items: center;
`
