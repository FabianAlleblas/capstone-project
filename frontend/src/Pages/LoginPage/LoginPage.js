import LoginForm from '../../Components/Forms/LoginForm'
import { ReactComponent as AppLogo } from '../../assets/icons/urbanplantslogo.svg'
import styled from 'styled-components/macro'
import ErrorModal from '../../Components/Modals/ErrorModal'

export default function LoginPage({
  userRegistration,
  userLogin,
  userData,
  setUserData,
}) {
  return (
    <PageWrapper>
      <AppLogo />
      <LoginForm userRegistration={userRegistration} userLogin={userLogin} />
      {userData?.error && (
        <ErrorModal setUserData={setUserData}>{userData.error}</ErrorModal>
      )}
    </PageWrapper>
  )
}

const PageWrapper = styled.section`
  background-color: var(--primary-dark);
  display: grid;
  height: 100vh;
  padding: 30px;
  place-items: center;
`
