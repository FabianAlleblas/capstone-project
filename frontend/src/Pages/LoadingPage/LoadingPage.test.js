import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import 'jest-styled-components'
import { MemoryRouter } from 'react-router-dom'
import LoginPage from './LoadingPage'
import LoadingPage from './LoadingPage'

describe('LoadingPage', () => {
  it('renders correctly', () => {
    const { container } = render(<LoginPage />)
    expect(container).toMatchSnapshot()
  })
})
