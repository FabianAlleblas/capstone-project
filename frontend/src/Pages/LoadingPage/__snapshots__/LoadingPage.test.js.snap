// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage renders correctly 1`] = `
.c1 {
  height: 200px;
}

.c2 {
  -webkit-animation: rotate 2s linear infinite;
  animation: rotate 2s linear infinite;
  height: 100px;
  width: 100px;
}

.c2 .path {
  -webkit-animation: dash 1.5s ease-in-out infinite;
  animation: dash 1.5s ease-in-out infinite;
  stroke-linecap: round;
  stroke: var(--primary-light);
}

.c0 {
  background-color: var(--primary-dark);
  display: grid;
  height: 100vh;
  place-items: center;
}

<div>
  <section
    class="c0"
  >
    <svg>
      urbanplantslogo.svg
    </svg>
    <div
      class="c1"
    >
      <svg
        class="c2"
        viewBox="0 0 50 50"
      >
        <circle
          class="path"
          cx="25"
          cy="25"
          fill="none"
          r="20"
          stroke-width="4"
        />
      </svg>
    </div>
  </section>
</div>
`;
